File: InventoryManager.Angular/inventory-manager-frontend/src/app/models/item-schema.ts
export class ItemSchema {
  id: string;
  name: string;
  properties: ItemSchemaProperty[];

  public constructor(init?: Partial<ItemSchema>) {
    Object.assign(this, init);
  }
}

export class ItemSchemaProperty {
  name: string;
  type: ItemSchemaPropertyType;
}

export enum ItemSchemaPropertyType {
  Text = 0,
  Number = 1
}

File: dotnet-backend/Models/ItemSchema.cs
using System.Collections.Generic;

namespace InventoryManager.Api.Models
{
    public class ItemSchema : OwnedEntity
    {
        public string Name { get; set; }
        public ICollection<ItemSchemaProperty> Properties { get; set; }
    }

    public class ItemSchemaProperty
    {
        public string Name { get; set; }
        public ItemSchemaPropertyType Type { get; set; }
    }

    public enum ItemSchemaPropertyType
    {
        Text = 0,
        Number = 1,
        Checkbox = 2
    }
}
